### ISSUE TITLE
Inserter: Its not obvious how to close the inserter

### ISSUE THREAD

**Issue by shaunandrews** [WordPress/gutenberg#22871]
During some recent usability testing, I watched a user who had managed to open both the inserter and the inspector at the same time. Her screen looked something like this:

<img width="1573" alt="image" src="https://user-images.githubusercontent.com/191598/83688180-fb7d1480-a5ba-11ea-8446-aad306563caf.png">

Now this person _had_ opened both of these panels intentionally, and was making some progress on creating a new page. However, with both panels open much of their content was "squished" and they wanted to hide some tools that they no longer needed.

This person quickly found the "x" icon in the top-right of the Inspector panel...

<img width="301" alt="image" src="https://user-images.githubusercontent.com/191598/83688320-33845780-a5bb-11ea-8f7e-ec39101c2c4f.png">

...but they were unable to find a way to close the inserter. You'll notice that there's no obvious way to close the panel.

<img width="369" alt="image" src="https://user-images.githubusercontent.com/191598/83688359-439c3700-a5bb-11ea-9304-37577d90d604.png">

The changing color of the button (from blue to black) wasn't obvious to them, and they didn't connect it with a _toggle_ interaction.

**Comment by paaljoachim** [comment]
A quick idea would actually be to turn the + icon into an x icon. Like so.
![Close-inserter-panel-X-bigger-screen](https://user-images.githubusercontent.com/5323259/83721478-6ec66b00-a63b-11ea-99ae-e77e7d2a7e12.jpg)


![Close-inserter-panel-X](https://user-images.githubusercontent.com/5323259/83721487-72f28880-a63b-11ea-93fd-e9ad427a2306.jpg)

As the X gives the idea of closing.

**Comment by shaunandrews** [comment]
That’s an interesting idea. However, part of me thinks that whatever we do, the pattern for closing a panel should be consistent regardless of if the panel is on the left or the right.

**Comment by chrisvanpatten** [comment]
It might be interesting to swap the search box down into the tab panel, then align the tab header styles in the inserter and document inspector. Right now they use different styles, but it might be a good opportunity to unify…

<img width="1183" alt="Screen Shot 2020-06-04 at 9 09 29 AM" src="https://user-images.githubusercontent.com/1231306/83760561-1c894800-a643-11ea-940c-a8687176232c.png">

(I don't necessarily love the grey background but thought I'd put it in to show what a 100% style alignment would look like)

**Comment by pablohoneyhoney** [comment]
Liking @paaljoachim idea, but it's worth looking first at other instances that would have that behavior and value if it still makes sense, to @shaunandrews point. 

I don't think we should copy the existing pattern in the sidebar, as this interface needs work to do and it would be a regression to adopt something that might evolve anyways. 

The search below the tabs is misleading, as the search has a higher architectural significance.

**Comment by ZebulanStanphill** [comment]
@pablohoneyhoney Well, the search bar actually changes behavior depending on which tab you're on, so it's really two different search bars. So it actually makes more sense for it to appear below the tabs.

**Comment by richtabor** [comment]
Is there any reason why the library/sidebar toggle acts differently from the settings sidebar toggle (OFF: icon without background color, ON: Icon with black background color)? 

Here's a Figma prototype of what it would feel like if they acted the same: https://www.figma.com/proto/VIyk72dx50GifV5ifPM63X/Library-Toggle?node-id=1%3A4&viewport=380%2C1085%2C0.5&scaling=contain

**Comment by ZebulanStanphill** [comment]
The "Add block" button is a primary button (just like the Publish button), so that's why it has a blue background by default. It is a bit awkward that the active state looks the same as a standard button, though.

**Comment by richtabor** [comment]
Why is it considered a primary button? 🤔

**Comment by ZebulanStanphill** [comment]
Because adding blocks was considered a primary action. I think @jasmussen would know more.

**Comment by pablohoneyhoney** [comment]
I don't think it's a primary action but intrinsic to the editing experience. Color and shape helps for hierarchy in the context of the collective UI, both within the top bar left functions as well as the rest of the interface.

**Comment by paaljoachim** [comment]
This is a somewhat associated issue. This is in regards to the inserter parent-child relationship in the buttons. Check out the video as well. https://github.com/WordPress/gutenberg/issues/21509#issuecomment-638195901

**Comment by mapk** [comment]
I'm favorable to @paaljoachim's idea about turning the `+` into an `x`. Noting that most of the topbar icons work as a toggle anyways, I think this could work well.

**Comment by webmaster-cob** [comment]
It never occurred to us to look outside of the panel for the close button. The only workaround we figured out was to click to the black plus icon on the page for inserting new blocks, not the plus button on the top toolbar. We also did a search online to figure out if there was a way to close it and didn't readily find one. Now that we know about the toggle on the top toolbar, we have a decent workaround. However, after reading the details in this ticket, it would make sense to either have a close X on the Browse All panel or remove the close X on the other panels and force users to open and close using the top toolbar. I wonder if there are any accessibility implications of not having an option to close panels easily from within the panel though.

**Comment by ZebulanStanphill** [comment]
#24429 changed the inserter behavior so that it automatically closes when focus leaves it (making it act more like a popover/dropdown), so I think this issue can be closed.

Remaining accessibility issues with the inserter are being discussed in #24975.

**Comment by afercia** [comment]
This was also reported on Trac (see https://core.trac.wordpress.org/ticket/50725) and during today's accessibility meeting we closed the Trac ticket in favor of this issue. I'm not sure the problem can be considered fully solved or at least I'd like to see it discussed more in depth by the accessibility team. Also, to keep track of the specific issue reported on Trac, I'd like to reopen this 🙂

**Comment by sarahricker** [comment]
This issue was discussed again in today's Accessibility + Design Office hours in [slack](https://wordpress.slack.com/archives/C02RP4X03/p1613584033371900).

Overall, we agreed that either design would work, and that it would be great to offer easier visibility on how to close the inserter. 

When opening the inserter:
- converting the `[+]` button that opens inserter into a `[x]` close inserter button would be helpful for users with dexterity issues that opened the inserter by accident, and which to close right away. 
- adding the `[x]` button to the right of the tab panel is also a great idea -- as it matches the other closable panels, adding consistency to the experience.

Ideal combination of options based on our discussion:
1. Use both, if not too complex! 😂  🥇  
2.  Use tab panel option only, _but convert the `[+]` button to be disabled_, as it doesn't actually mean anything at all if the inserter is already open.
3. Convert the `[+]` button that opens inserter into a `[x]` close inserter button only. (seems the easiest answer, but prefer option 1). 

Any of those would be acceptable re: accessibility. Thanks so much @shaunandrews! 🎉

**Comment by paaljoachim** [comment]
Here is a support issue that was shared in the slack design channel.
https://wordpress.org/support/topic/stop-gutenberg-sidebar-disappearing-when-edit-code-in-editor/

"When I click to edit any part of the editor text or code, the gutenberg sidebar disappears, now I know this is new update so it might be normal to work that but it is really annoying that every time I click to write paragraph text, whole page move to the side.
Please take a look at the photos, Is it possible to keep the sidebar sticky all the time."


![Block-Inserter-Gutenberg](https://user-images.githubusercontent.com/5323259/110697029-159fef00-81ec-11eb-9f24-fe8c816de75b.jpg)

**Comment by bcworkz-wp** [comment]
When opening the inserter, the right side meta boxes close (5.7). This makes sense due to limited screen real estate. However, when the inserter closes, the previously open meta boxes remain closed. This isn't cool, the user's previous state should be restored. If it was closed, it stays closed. If it was open, it should be reopened.

**Comment by paaljoachim** [comment]
Hey @bcworkz-wp 
I believe you are talking about what is mentioned in this issue: 
**If the Inserter panel closes the Inspector panel, the inspector should re-open**
https://github.com/WordPress/gutenberg/issues/23968

**Comment by paaljoachim** [comment]
@jameskoster 
James has made a PR for an open and close "Add block" inserter panel.
Here: https://github.com/WordPress/gutenberg/pull/29759

**Comment by annezazu** [comment]
Closing this out now that Jay's PR has merged and this no longer occurs 🥳 


https://user-images.githubusercontent.com/26996883/118590545-db108080-b75f-11eb-9834-cfd06793f1fa.mov
